### ISSUE TITLE
How to use global markupOnly option?

### ISSUE THREAD

**Issue by axelmm** [edvardchen/eslint-plugin-i18next#34]
I have a configuration problem ... not being eslint expert 
... I don't know how/where in `.eslintrc.js` it has to be defined.

I work with react/JSX using f.e. css in js  - literal values gives false positives, too.
Is there a possibility to define `ignoreAttribute` option globally (.eslintrc.js), not per line?

**Comment by edvardchen** [comment]
@axelmm  try
```json
{
  "rules": {
    "i18next/no-literal-string": [2, { "markupOnly": true }]
  }
}

```

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for <Fragment/>), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for < Fragment/ >), plugin doesn't see this in markupOnly mode.

**Comment by axelmm** [comment]
@edvardchen
Thanks for your help ... and work on really usable tool.

I guess there is no global `ignoreAttribute` option ... and should use 

        'i18next/no-literal-string': [2, {"onlyAttribute": []} ],

... BTW in react you can use `<>test</>` markup (shortcut for `<Fragment/>`), plugin doesn't see this in markupOnly mode.

**Comment by edvardchen** [comment]
good to hear feedback. i will fix it soon.

ps: only attribute option actually turn on markupOnly

**Comment by axelmm** [comment]
it works but reported twice - acceptable

I know that `onlyAttribute` turns on `markupOnly` ... I'm using it [undocummented usage] as [IMHO missing] `ignoreAllAttributes` (can be implemented as alias?).

**Comment by axelmm** [comment]
... the trick didn't work in other config (docker/vscode/pnpify), I had to use
  
       'i18next/no-literal-string': [2, {"onlyAttribute": [""]} ],

**Comment by axelmm** [comment]
@edvardchen
Next "missing" in markup mode, strings within code fragments 'evals'/conditionals not recognized, f.e.

``          <FormHelperText>
            {negative ? "Block heights" : "Well depths"}
             {someValue + " some text/unit"}
          </FormHelperText>

**Comment by axelmm** [comment]
@edvardchen
Next "missing" in markup mode, strings within code fragments 'evals'/conditionals not recognized, f.e.

``          <FormHelperText>
``            {negative ? "Block heights" : "Well depths"}
``            {someValue + " some text/unit"}
``          </FormHelperText>

**Comment by axelmm** [comment]
@edvardchen
Next "missing" in markup mode, strings within code fragments 'evals'/conditionals not recognized, f.e.

          <FormHelperText>
            {negative ? "Block heights" : "Well depths"}
            {someValue + " some text/unit"}
          </FormHelperText>

**Comment by edvardchen** [comment]
> ... the trick didn't work in other config (docker/vscode/pnpify), I had to use
> 
> ```
>    'i18next/no-literal-string': [2, {"onlyAttribute": [""]} ],
> ```

I misunderstood you. The `makrupOnly` option still will validate the jsx attritubes

**Comment by edvardchen** [comment]
> @edvardchen
> Next "missing" in markup mode, strings within code fragments 'evals'/conditionals not recognized, f.e.
> 
> ```
>       <FormHelperText>
>         {negative ? "Block heights" : "Well depths"}
>         {someValue + " some text/unit"}
>       </FormHelperText>
> ```

I will fix it

**Comment by edvardchen** [comment]
See [the discussion](https://github.com/edvardchen/eslint-plugin-i18next/issues/17). The `makrupOnly` would only validate the JSX text and JSX attritubes exclude the expressions. 

Since I don't use this mode in my projects, I am not sure what the suitable behavors should be. 

You can fork and change it to satisfy your requirement.
